mlflow/store/_unity_catalog/oss_registry/oss_rest_store.py

```diff
@@ -0,0 +1,136 @@
+import functools
+
+from mlflow.protos.unity_catalog_oss_messages_pb2 import (
+    CreateRegisteredModel,
+    RegisteredModelInfo,
+    TagKeyValue,
+)
+from mlflow.protos.unity_catalog_oss_service_pb2 import UnityCatalogService
+from mlflow.store.model_registry.base_rest_store import BaseRestStore
+from mlflow.utils._oss_unity_catalog_utils import registered_model_from_uc_oss_proto
+from mlflow.utils._unity_catalog_utils import get_full_name_from_sc
+from mlflow.utils.annotations import experimental
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import (
+    _REST_API_PATH_PREFIX,
+    extract_all_api_info_for_service,
+    extract_api_info_for_service,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UnityCatalogService, _REST_API_PATH_PREFIX)
+_METHOD_TO_ALL_INFO = extract_all_api_info_for_service(UnityCatalogService, _REST_API_PATH_PREFIX)
+
+
+@experimental
+class OssUnityCatalogStore(BaseRestStore):
+    """
+    Client for an Open Source Unity Catalog Server accessed via REST API calls.
+    """
+
+    def __init__(self, store_uri):
+        super().__init__(get_host_creds=functools.partial(get_databricks_host_creds, store_uri))
+
+    def _get_response_from_method(self, method):
+        method_to_response = {
+            CreateRegisteredModel: CreateRegisteredModel.Response,
+        }
+        return method_to_response[method]()
+
+    def _get_endpoint_from_method(self, method):
+        return _METHOD_TO_INFO[method]
+
+    def _get_all_endpoints_from_method(self, method):
+        return _METHOD_TO_ALL_INFO[method]
+
+    def create_registered_model(self, name, tags=None, description=None):
+        full_name = get_full_name_from_sc(name, None)
+        [catalog_name, schema_name, model_name] = full_name.split(".")
+        comment = description if description else ""
+        tags = [TagKeyValue(key=tag.key, value=tag.value) for tag in (tags or [])]
+        req_body = message_to_json(
+            CreateRegisteredModel(
+                registered_model_info=RegisteredModelInfo(
+                    name=model_name,
+                    catalog_name=catalog_name,
+                    schema_name=schema_name,
+                    comment=comment,
+                    tags=tags,
+                )
+            )
+        )
+        response_proto = self._call_endpoint(CreateRegisteredModel, req_body)
+        return registered_model_from_uc_oss_proto(response_proto.registered_model_info)
+
+    def update_registered_model(self, name, description):
```

Comment:
Is this first draft of the pr, and we'll implement the other methods in a follow up pr?

Reply:
I will let @rohitarun-db implement the rest of these as part of his starter project.